builtin/trap_osh.py

```diff
@@ -49,6 +49,7 @@ def __init__(self, signal_safe):
         self.signal_safe = signal_safe
         self.hooks = {}  # type: Dict[str, command_t]
         self.traps = {}  # type: Dict[int, command_t]
+        self.ignored_sigs = {}  # type: Dict[int, bool]  # Signals set to SIG_IGN
```

Comment:
Hm instead of a separate dict, I wonder if we should reuse `command.NoOp`

That is defined in `frontend/syntax.asdl`, and then right now we use it for empty commands like `>out.txt`

osh/cmd_parse.py:

```
         if len(words) == 0:  # e.g.  >out.txt  # redirect without words
             assert len(redirects) != 0
             if typed_loc is not None:
                 p_die("Unexpected typed args", typed_loc)
             return command.Redirect(command.NoOp, redirects)
```